sycl/include/sycl/detail/pi.hpp

```diff
@@ -162,128 +70,23 @@ void *loadOsLibrary(const std::string &Library);
-// TODO: GlobalPlugin does not seem to be needed anymore. Consider removing it!
-extern std::shared_ptr<plugin> GlobalPlugin;
-
-// Performs PI one-time initialization.
-std::vector<PluginPtr> &initialize();
+// Performs UR one-time initialization.
+std::vector<PluginPtr> &initializeUr();
 
 // Get the plugin serving given backend.
 template <backend BE> __SYCL_EXPORT const PluginPtr &getPlugin();
 
-// Utility Functions to get Function Name for a PI Api.
-template <PiApiKind PiApiOffset> struct PiFuncInfo {};
-
-#define _PI_API(api)                                                           \
-  template <> struct PiFuncInfo<PiApiKind::api> {                              \
-    using FuncPtrT = decltype(&::api);                                         \
-    inline const char *getFuncName() { return #api; }                          \
-    inline FuncPtrT getFuncPtr(PiPlugin MPlugin) {                             \
-      return MPlugin.PiFunctionTable.api;                                      \
-    }                                                                          \
-  };
-#include <sycl/detail/pi.def>
-
-/// Emits an XPTI trace before a PI API call is made
-/// \param FName The name of the PI API call
-/// \return The correlation ID for the API call that is to be used by the
-/// emitFunctionEndTrace() call
-uint64_t emitFunctionBeginTrace(const char *FName);
-
-/// Emits an XPTI trace after the PI API call has been made
-/// \param CorrelationID The correlation ID for the API call generated by the
-/// emitFunctionBeginTrace() call.
-/// \param FName The name of the PI API call
-void emitFunctionEndTrace(uint64_t CorrelationID, const char *FName);
-
-/// Notifies XPTI subscribers about PI function calls and packs call arguments.
-///
-/// \param FuncID is the API hash ID from PiApiID type trait.
-/// \param FName The name of the PI API call.
-/// \param ArgsData is a pointer to packed function call arguments.
-/// \param Plugin is the plugin, which is used to make call.
-uint64_t emitFunctionWithArgsBeginTrace(uint32_t FuncID, const char *FName,
-                                        unsigned char *ArgsData,
-                                        pi_plugin Plugin);
-
-/// Notifies XPTI subscribers about PI function call result.
-///
-/// \param CorrelationID The correlation ID for the API call generated by the
-/// emitFunctionWithArgsBeginTrace() call.
-/// \param FuncID is the API hash ID from PiApiID type trait.
-/// \param FName The name of the PI API call.
-/// \param ArgsData is a pointer to packed function call arguments.
-/// \param Result is function call result value.
-/// \param Plugin is the plugin, which is used to make call.
-void emitFunctionWithArgsEndTrace(uint64_t CorrelationID, uint32_t FuncID,
-                                  const char *FName, unsigned char *ArgsData,
-                                  pi_result Result, pi_plugin Plugin);
-
 /// Tries to determine the device binary image foramat. Returns
```

Comment:
This will be completely removed, we still need to port over the device binary objects/functions and the mock tests before we can do this. It will be done as part of this PR.

Reply:
the device binary stuff is still in ur.hpp but it has been renamed